docs/DeveloperGuide.md

```diff
@@ -347,17 +347,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
 Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
-`* * *` | user | add a new person |
-`* * *` | user | delete a person | remove entries that I no longer need
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
-
-{More to be added}
+`* * *` | user | add a new task with completion date | remind myself of things I have to do and the deadlines to meet
+`* * *` | user | add a new [recurring task](#recurring-task) | prevent the need of keying the same cyclical task every period
+`* * *` | user | see all tasks | have an overview of all the things I need to do and decide what I should do first
+`* * *` | user | update a task | change entries that are errorneous or outdated
+`* * *` | user | mark a task as completed/incompleted | focus on the tasks I have still not cleared
+`* * *` | user | delete a task | remove entries that I no longer need
+`* * *` | user | find a task by name | locate details of task without having to go through the entire list
+`* * *` | user | find a task by tag | locate details of task without having to go through the entire list
+`* * *` | user | add tag to task | group my tasks to provide more context
+`* * *` | user | update tag in task | change entries that are errorneous or outdated
+`* * *` | user | remove tag from task | remove entries that I no longer need
+`* * *` | user | undo previous command(s) | recover gracefully from making mistakes
+`* * *` | user | redo previously undone command(s) | recover gracefully from  wrongly resolving a mistake
+`* * *` | user | see my [command history](#command-history) | to identify what I have to recover if I accidentally performed some wrong commands.
+`* * *` | user | change my storage file path | decide where I want to save my files for my own use
+`* *` | user with many tasks | sort task by [priority level](#priority-level) then by end date | figure out which task should be cleared first
+`* *` | user | add an [alias](#alias) for a command | customise my own keyboard shortcuts to improve my efficiency
+`* *` | user | update an alias for a command | change entries that are errorneous or outdated
+`* *` | user | view all alias for commands | review the alias in case I forget what I set for them
 
```

Comment:
Exiting program and clear all tasks is kinda.. trivial right.
As the user, I should be able to exit the program, so that I can...?
As the user, I should be able to clear all task, so that I can...?

I will update the delete alias user story.